### ISSUE TITLE
Internal CatBoost Error: BitsPerKey 16 exceeds maximum width 8

### ISSUE THREAD

**Issue by Pascal-Iv** [catboost/catboost#1010]
Problem:
I am fitting a model and after it is done I get the error. I only get the error when using a large number of iterations.

**My code to fit the model:**
catboost_model = catboost.CatBoostClassifier(**bestparam)
catboost_model.fit(train_pool, eval_set=eval_pool, plot=False, verbose=20, use_best_model=True)
**Parameter:**
{'depth': 10, 'iterations': 2500, 'learning_rate': None, 'l2_leaf_reg': 10, 'border_count': 255, 'scale_pos_weight': 13.903414634146342, 'task_type': 'GPU', 'od_type': 'Iter', 'od_wait': 40, 'loss_function': 'Logloss', 'eval_metric': 'AUC'}

I work on Google Collab with GPU and my catboost version is '0.17.3'

**The stack trace**

CatBoostError                             Traceback (most recent call last)
<ipython-input-26-a437d1a764a1> in <module>()
      1 catboost_model = catboost.CatBoostClassifier(**bestparam)
----> 2 catboost_model.fit(train_pool, eval_set=eval_pool, plot=False, verbose=20, use_best_model=True)

2 frames
/usr/local/lib/python3.6/dist-packages/catboost/core.py in fit(self, X, y, cat_features, sample_weight, baseline, use_best_model, eval_set, verbose, logging_level, plot, column_description, verbose_eval, metric_period, silent, early_stopping_rounds, save_snapshot, snapshot_file, snapshot_interval, init_model)
   3676         self._fit(X, y, cat_features, None, sample_weight, None, None, None, None, baseline, use_best_model,
   3677                   eval_set, verbose, logging_level, plot, column_description, verbose_eval, metric_period,
-> 3678                   silent, early_stopping_rounds, save_snapshot, snapshot_file, snapshot_interval, init_model)
   3679         return self
   3680 

/usr/local/lib/python3.6/dist-packages/catboost/core.py in _fit(self, X, y, cat_features, pairs, sample_weight, group_id, group_weight, subgroup_id, pairs_weight, baseline, use_best_model, eval_set, verbose, logging_level, plot, column_description, verbose_eval, metric_period, silent, early_stopping_rounds, save_snapshot, snapshot_file, snapshot_interval, init_model)
   1609                 params,
   1610                 allow_clear_pool,
-> 1611                 train_params["init_model"]
   1612             )
   1613 

/usr/local/lib/python3.6/dist-packages/catboost/core.py in _train(self, train_pool, test_pool, params, allow_clear_pool, init_model)
   1168 
   1169     def _train(self, train_pool, test_pool, params, allow_clear_pool, init_model):
-> 1170         self._object._train(train_pool, test_pool, params, allow_clear_pool, init_model._object if init_model else None)
   1171         self._set_trained_model_attributes()
   1172 

_catboost.pyx in _catboost._CatBoost._train()

_catboost.pyx in _catboost._CatBoost._train()

CatBoostError: catboost/libs/algo/index_hash_calcer.h:82: Internal CatBoost Error (contact developers for assistance): BitsPerKey 16 exceeds maximum width 8

**Comment by annaveronika** [comment]
Thank you for the report, we are looking into this!

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Hi Pascal,
Could you please share the dataset to reproduce the issue?
With random data (10M rows, 100 columns), CatBoost works fine.

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
As a workaround, could you please try border_count=254?
It looks like the bug is caused by https://github.com/catboost/catboost/blob/15295dbbaf4a136b28e5658d066a19b907a521e9/catboost/libs/quantization/utils.h#L153

**Comment by Pascal-Iv** [comment]
Thank you very much for getting back to me. It works with the workaround you proposed. Do you still want me to share the data?

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Not at the moment.
If fixing the issue above does not fix the error you see, then we will ask for a reproducer :)

**Comment by pseudotensor** [comment]
I also see this, trying work-around...

**Comment by kizill** [comment]
Do you use 0.17.4 version?

**Comment by pseudotensor** [comment]
Yes sir.
```
total 76
drwx------  2 jon jon 4096 Sep 25 02:15 category_encoders-2.0.0.dist-info/
drwx------  3 jon jon 4096 Sep 25 02:15 ../
drwx------  5 jon jon 4096 Sep 25 02:15 examples/
drwx------  4 jon jon 4096 Sep 25 02:15 category_encoders/
-rw-------  1 jon jon 9955 Oct  3 00:22 retrying.py
drwx------  4 jon jon 4096 Oct  3 00:22 _plotly_utils/
drwx------  2 jon jon 4096 Oct  3 00:22 catboost-0.17.4.dist-info/
drwx------  5 jon jon 4096 Oct  3 00:22 catboost/
drwx------  2 jon jon 4096 Oct  3 00:22 __pycache__/
drwx------  2 jon jon 4096 Oct  3 00:22 graphviz-0.13.dist-info/
drwx------  3 jon jon 4096 Oct  3 00:22 graphviz/
drwx------ 13 jon jon 4096 Oct  3 00:22 plotly/
drwx------  2 jon jon 4096 Oct  3 00:22 retrying-1.3.3.dist-info/
drwx------  4 jon jon 4096 Oct  3 00:22 plotlywidget/
drwx------  3 jon jon 4096 Oct  3 00:22 _plotly_future_/
drwx------  2 jon jon 4096 Oct  3 00:22 plotly-4.1.1.dist-info/
drwx------ 16 jon jon 4096 Oct  3 00:22 ./
```

**Comment by pseudotensor** [comment]
work-around seem to work so far.

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
Dear @pseudotensor !
This issue should have been fixed in CatBoost version 0.17.5. Could you please check?
E.
